src/mame/drivers/pacman.cpp

```diff
@@ -7648,6 +7761,18 @@ void pacman_state::init_pengomc1()
 		romdata[i] = buf[i^0xff];
 }
 
+uint8_t pacman_state::clubpacm_input_r(offs_t offset)
+{
+	uint8_t data = ioport((offset & 0x40) ? "IN1" : "IN0")->read();
```

Comment:
It’s not acceptable to continue adding runtime tag map lookups like this.  Use a `required_ioport_array` instead.

Reply:
Done. Also moved the new stuff into `class clubpacm_state` and cleaned up the IN0/IN1 mapping a little bit.